### DISCUSSION TITLE
Passing exception from C# to C++ in swig directors

### DISCUSSION THREAD

**Discussion by adkg** [swig/swig#2063] [General]
I have seen post where people have achieved using %feature("director:except") but not sure if its really supported for C# or not.

Scenario: I have delegate exposed in C++ layer which i am exposing in C# through swig directors and doing some conversion.
I need a way where if C# delegate implementation(customer side) , i want to pass it back to C++ so that i can make a decision whether to proceed or throw it back to C#.

**Comment by ojwb** [comment]
> %feature("director:except") but not sure if its really supported for C# or not.

It's not documented in the C# chapter of the manual, and there's no code to handle it in `Source/Modules/csharp.cxx` (unlike for other target languages which do document it) so it seems it's not supported for C# currently.
